src/sql/workbench/contrib/connection/browser/connectionStatus.ts

```diff
@@ -56,19 +62,62 @@ export class ConnectionStatusbarItem extends Disposable implements IWorkbenchCon
 		this._statusbarService.updateEntryVisibility(ConnectionStatusbarItem.ID, true);
 	}
 
-	// Update the connection status shown in the bar
+	/**
+	 * Update the connection status shown in the bar
+	 */
 	private _updateStatus(): void {
-		let activeConnection = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
+		let activeConnection: IConnectionProfile = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
+		let connectionId: string = undefined;
+		let activeEditorInfo: ConnectionManagementInfo = this.getCurrentActiveEditorInfo();
+		// retrieve the currently focused editor so we can get the correct connection id value for it.
+		if (activeEditorInfo) {
+			// We only want to display the connection ID for editor connections, so first check to make
+			// sure the connection we're displaying information for (which may be from OE or other
+			// sources) matches the current active editor before updating the ID
+			if (activeEditorInfo && activeEditorInfo?.connectionProfile?.id === activeConnection?.id) {
+				connectionId = activeEditorInfo?.serverConnectionId;
+			}
+		}
 		if (activeConnection) {
-			this._setConnectionText(activeConnection);
+			this._setConnectionText(activeConnection, connectionId)
 			this.show();
-		} else {
+		}
+		else {
 			this.hide();
 		}
 	}
 
-	// Set connection info to connection status bar
-	private _setConnectionText(connectionProfile: IConnectionProfile): void {
+	/**
+	 * Updates the connection server id after a query has been run, since the URI may have changed.
+	 * If the current editor is the one being updated, we will do a refresh if the id is different.
+	 */
+	private _refreshQueryUri(uriToUpdate: string, idForUpdate: string) {
```

Comment:
is the comment and method name actually accurate? you are just updating the status here, not the URI of the query.

I feel like this method is handling things more than it should. update the connection id should be handled in the connection management code, I believe this file is for the UI related logic.